aten/src/ATen/native/vulkan/api/Adapter.h

```diff
@@ -0,0 +1,36 @@
+#pragma once
+
+#include <ATen/native/vulkan/api/Common.h>
+#include <ATen/native/vulkan/api/Runtime.h>
+
+namespace at {
+namespace native {
+namespace vulkan {
+namespace api {
+
+//
+// A Vulkan Adapter represents a physical device and its properties.  Adapters
+// are enumerated through the Runtime and are used in creation of Contexts.
+// Each tensor in PyTorch is associated with a Context to make the
+// device <-> tensor affinity explicit.
+//
+
+struct Adapter final {
+  Runtime* runtime;
```

Comment:
Thinking about if reference can be used here.

Could it be nullptr here for some cases?

Reply:
Great point, thank you.  Reference members make classes uncopyable, which is the reason behind my use of pointers, but I actually never checked to see if it is OK for Adapter to be non-copyable.  I'll make this change and see if I can make it work.  Thanks.

Reply:
To report back on this - I tried converting the pointers to references both here and on `api::GPU` but that makes them makes them non-move-assignable (can move construct though) besides non-copyable which means objects of these types cannot be returned from functions without mandatory copy elision introduced in C++-17.  :/